wire-runtime/src/nativeMain/kotlin/com/squareup/wire/ProtoAdapter.kt

```diff
@@ -24,8 +24,13 @@ actual abstract class ProtoAdapter<E> actual constructor(
   internal actual val fieldEncoding: FieldEncoding,
   actual val type: KClass<*>?
 ) {
-  internal actual var packedAdapter: ProtoAdapter<List<E>>? = null
-  internal actual var repeatedAdapter: ProtoAdapter<List<E>>? = null
+  // Using by lazy doesn't work on Native since it creates cyclic refs between objects
+  // (commonCreateXxx() is an extension function on ProtoAdapter), so we can't cache adapters here
+  // and will have to recreate them every time.
```

Comment:
is this expensive?

Reply:
Encoding and decoding will both be creating new instances with this approach, plus repeated and packed adapters for common field types will be recreated, so if your proto has 10 `repeated int32` fields there will be 10 allocations during encoding and 10 during decoding. This sucks. Alternatively we can eagerly create those adapters on Native.